### ISSUE TITLE
Update the grid.json according to the version 2.1.0

### ISSUE THREAD

**Issue by agrignard** [CityScope/CSL_Lima#10]
@vaap1997 we are working on a standard format that any reader/writer of a grid can use. 

The specification is here
https://github.com/CityScope/CS_CityIO_Backend/wiki/Data-Format

and you can find an example of a table that is compatible with this here:
https://cityio.media.mit.edu/api/table/virtual_table

it would be great if this file https://github.com/CityScope/CSL_Lima/blob/master/projects/LegoReader/data/grid.json is following the patterns.

You are not that far actually what is missing is 

- the header
https://github.com/CityScope/CS_CityIO_Backend/wiki/Data-Format#header-dictionary

- the meta (ideally you shouldn't worry to much as later it will be filled by the cityIO server)
https://github.com/CityScope/CS_CityIO_Backend/wiki/Data-Format#meta

- the grid information is here https://github.com/CityScope/CS_CityIO_Backend/wiki/Data-Format#grid2d-list
you just need to store the grid value in this format for each cell you just have to send the rot and the id (the color pattern is not usefull in this case). and then for the grid the format is slightly different (we don't store anymore the x and y we just store all the cell as a list where the first element of the grid is cell 0,0, the second cell 0,1 etc..

So in your case we will have 
`````
"block":["type","rotation"]
`````

and a list of cell value with the id and the rotation

`````
e.g [180,1],[90,1],[270,3], etc
`````

**Comment by vaap1997** [comment]
Arnaud, I just change the grid format

**Comment by agrignard** [comment]
@vaap1997 this is getting there. In the grid you are now sending mor eor less what we need (even if the color pattern are useless I think)

In order to be really compatible with 2.10 you will need also to add the header.

I will provide you an example soon the header is bascially a kind of templat that is suppose to put the information relative to the table.

**Comment by agrignard** [comment]
This is how your header could look (for sure this will depend on the table that is scanning) but let's imagine you have a table in LIMA the header might look like this:

`````
"header":{"name":"cityscope_lima","spatial":{"nrows":16,"ncols":16,"physical_longitude":-77,"physical_latitude":12,"longitude":-77,"latitude":12,"cellSize":10,"rotation":0},"owner":{"name":"Vanessa","title":"Researcher,"institute":"univeristy of Lima"},"block":["type","height","rotation"],"mapping":{"type":{"0":"RL","1":"RM","2":"RS","3":"OL","4":"OM","5":"OS","6":"ROAD"}}}
``````


Note that in the case of your generic tools the mapping is maybe optional as it here to say in some case that id 1 correspond to a certain type "RL" in our. But at least it might be something like


``````
{"0":"0","1":"1",....,"n":"n"}
```````

where n is number of type of block

**Comment by vaap1997** [comment]
I think is done! Tell me if you have any comments

**Comment by agrignard** [comment]
@JesusGarcia98 we use to have a feature that enables to export the grid in a grid.json format in order to be compatible with cityIO. In the master branch it's the method called exportGrid however it has disappear from the dev branche, any reason why? Would it be possible to put it back in the dev branch?

**Comment by JesusGarcia98** [comment]
I thought the exportGridUDP() method was the replacement of the method you mentioned, which is why I did not include it. I now have included it again in the dev branch, along with an update in the README.

**Comment by agrignard** [comment]
You're right for now there is 2 export (which ultimately should be the same) I will make an issue about this.

The export to UDP is a simple way (and lighter in term of size of the string) to exchange data the exportGrid is a way to be compatible with our other project using cityIO (maybe we should call it exporttoCityIO

**Comment by agrignard** [comment]
Seems to have been fixed in the dev branch with this commit
https://github.com/CityScope/CSL_Lima/commit/5443a66e88065c15f5c50322a630810d2478aa74

**Comment by agrignard** [comment]
I repopen this issue has the exported format is not totally compatible with the 2.1 format (the way the grid are stored)

**Comment by agrignard** [comment]
See https://github.com/CityScope/CS_CityIO_Backend/commit/aee3e66a8387c0d1b2dd2ea84a00fa308175fcd7

**Comment by JesusGarcia98** [comment]
I updated the export method following the example you indicated.
